client/core/core.go

```diff
@@ -1008,13 +1011,58 @@ func (c *Core) connectedWallet(assetID uint32) (*xcWallet, error) {
 	return wallet, nil
 }
 
+func (c *Core) connectWallet(w *xcWallet) error {
+	err := w.Connect(c.ctx)
+	if err != nil {
+		return newError(walletErr, "error connecting %s wallet: %v", unbip(w.AssetID), err)
+	}
+	// If the wallet is not synced, start a loop to check the sync status until
+	// it is.
+	if !w.synced {
+		// If the wallet is shut down before sync is complete, exit the syncer
+		// loop.
+		innerCtx, cancel := context.WithCancel(c.ctx)
+		go func() {
+			w.connector.Wait()
+			cancel()
+		}()
+		// The syncer loop.
+		go func() {
+			ticker := time.NewTicker(syncTickerPeriod)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					synced, progress, err := w.SyncStatus()
+					if err != nil {
+						c.log.Errorf("error monitoring sync status for %s", unbip(w.AssetID))
+						return
+					}
+					w.mtx.Lock()
+					w.synced = synced
+					w.syncProgress = progress
+					w.mtx.Unlock()
+					c.notify(newWalletStateNote(w.state()))
+					if synced {
+						return
+					}
+
+				case <-innerCtx.Done():
+					return
+				}
+			}
+		}()
+	}
+	return nil
+}
+
 // Connect to the wallet if not already connected. Unlock the wallet if not
 // already unlocked.
 func (c *Core) connectAndUnlock(crypter encrypt.Crypter, wallet *xcWallet) error {
 	if !wallet.connected() {
-		err := wallet.Connect(c.ctx)
+		err := c.connectWallet(wallet)
 		if err != nil {
-			return newError(walletErr, "error connecting %s wallet: %v", unbip(wallet.AssetID), err)
+			return err
```

Comment:
Say the client does something like starts a swap and their computer restarts or something happens where they restart and a bitcoind is till syncing.  Even if that's not enough to take action we'd still want to unlock the wallet I think, but this won't unlock the wallet if it's still syncing.

How would a use handle this state?  What would the problem be to unlock the wallet regardless of sync status?

Reply:
Oh, I think I misread the flow.  `connectWallet` does NOT till returns a non-nil error for not syncd status, so it would still proceed to unlock, right?

Reply:
That's right. The sync loop is run in a goroutine. `connectWallet` is just a convenient place to start it.